c2rust-analyze/src/main.rs

```diff
@@ -456,6 +456,25 @@ fn run(tcx: TyCtxt) {
         gacx.fn_sigs.insert(ldid.to_def_id(), lsig);
     }
 
+    // Collect all `static` items.
+    let all_static_ldids = all_static_items(tcx);
+    eprintln!("statics:");
+    for &ldid in &all_static_ldids {
+        eprintln!("  {:?}", ldid);
+    }
+
+    // Assign global `PointerId`s for types of `static` items.
+    assert!(gacx.static_tys.is_empty());
+    gacx.static_tys = HashMap::with_capacity(all_static_ldids.len());
+    for &ldid in &all_static_ldids {
+        let did = ldid.to_def_id();
+        if !matches!(tcx.def_kind(did), DefKind::Static(..)) {
```

Comment:
given the definition of `all_static_items`, when would this not be true? could you cut out the middleman and return a list of `DefId`s instead of `LocalDefId`s that have `DefKind::Static`?

Reply:
Yep, will do.